Doc/c-api/tuple.rst

```diff
@@ -35,6 +35,10 @@ Tuple Objects
 
    Return a new tuple object of size *len*, or *NULL* on failure.
 
+   .. versionchanged:: 3.7
+
+      ``PyTuple_New(0)`` always succeeds and never returns *NULL*.
```

Comment:
Maybe it would be enough. But with this guarantee other users can get a benefit from this. I asked on Python-Dev whether it is good to add such promise.